doc/tutorial/read_mapper/index.md

```diff
@@ -0,0 +1,163 @@
+# Implementing your own read mapper with SeqAn3 {#tutorial_read_mapper}
+
+<b>Learning Objective:</b><br>
+In this tutorial you will learn how combine the taught components of previous tutorials to create your very first
+SeqAn3 application: a read mapper!
+
+\tutorial_head{High, 90 Minutes, All,}
+
+[TOC]
+
+<br><br>
+
+# Introduction
+
+Read mapping is a common task in bioinformatics and is often the first step of an in-depth analysis. Its aim is to
+identify positions where a query sequence (read) matches with up e errors a reference sequence. In this example we will
+implement a read mapper step by step and makes use of the previous tutorials.
+
+# Agenda
+
+* Parse arguments
+* Read input files
+* Search for approximate matches
+* Align the search results
+* Write final results into a SAM file
+
+# Step 1 - Parsing arguments
+As a first step, we want to parse arguments for our readmapper.
+
+\assignment{Parsing arguments}
+Write a program that parses arguments and print the parser arguments.
+
+You should parse:
+* The path to the reference file
+* The path to the query file
+* An output path
+* The maximum number of errors we want to allow (between 0 and 4)
+
+Follow the best practice and create:
+* A function `run_program` that prints the parsed arguments
+* A struct `cmd_arguments` that stores the arguments
+* A function `initialise_argument_parser` to add meta data and options to the parser
+* A `main` function that parses the arguments and calls `run_program`
+
+Use validators where useful!
+
+\hint
+Your `main` may look like this:
+\snippet doc/tutorial/read_mapper/read_mapper_step1.cpp main
+\endhint
+\endassignment
+\solution
+\include doc/tutorial/read_mapper/read_mapper_step1.cpp
+\endsolution
+
+# Step 2 - Reading the input
+Now we want to read the input reference and queries.
```

Comment:
As a next step, we want to use the parsed file names to read in our data. This will be done for the reference and query sequences using seqan3::sequence_file_input class. As a guide, you can take a look at the [Sequence I/O Tutorial](#tutorial_sequence_file).